lib/sgtn-client/api/translation.rb

```diff
@@ -59,6 +59,7 @@ def self.getString_f(component, key, args, locale, *optionals)
 
       def self.getStrings(component, locale)
         SgtnClient.logger.debug "[Translation][getStrings]component=#{component}, locale=#{locale}"
+        locale = SgtnClient::LocaleUtil.get_best_locale(locale)
         items = get_cs(component, locale)
         default = SgtnClient::Config.configurations.default
         if items.nil? || items["messages"] == nil
```

Comment:
```suggestion
        items = get_cs(component, locale) if locale != LocaleUtil.get_source_locale
        default = SgtnClient::Config.configurations.default
        if (defined?(items)).nil? || items.nil? || items["messages"] == nil
```

Reply:
Looks complex. If there isn't a bug, I prefer to perceed with current to make changes as small as possible.